src/imports.rs

```diff
@@ -478,7 +478,7 @@ impl UseTree {
 
         // Normalise foo::{bar} -> foo::bar
         if let UseSegment::List(ref list) = last {
-            if list.len() == 1 {
+            if list.len() == 1 && format!("{}", list.first().unwrap()) != String::from("self") {
```

Comment:
This could be simplified to the following code:
```suggestion
            if list.len() == 1 && format!("{}", list[0]) != "self" {
```

Reply:
Or:

```suggestion
            if list.len() == 1 && list[0].to_string() != "self" {
```